substrate/frame/support/src/traits/tokens/asset_ops/common_strategies.rs

```diff
@@ -0,0 +1,391 @@
+//! This modules contains the common asset ops strategies.
+
+use super::*;
+use codec::{Decode, Encode, MaxEncodedLen};
+use scale_info::TypeInfo;
+use sp_runtime::RuntimeDebug;
+
+/// The `CheckState` is a strategy that accepts an `Inspect` value and the `Inner` strategy.
+///
+/// It is meant to be used when the asset state check should be performed
+/// in addition to the `Inner` strategy.
+/// The inspected state must be equal to the provided value.
+///
+/// The `CheckState` implements all potentially state-mutating strategies that the `Inner`
+/// implements.
+pub struct CheckState<Inspect: InspectStrategy, Inner = Unchecked>(pub Inspect::Value, pub Inner);
```

Comment:
How about:
```suggestion
pub struct CheckState<Inspect: InspectStrategy, CheckStrategy = AlwaysValid>(pub Inspect::Value, pub Inner);
```
?

Reply:
I renamed the `Unchecked` strategy to `NoParams` since it's not about the check. It's about performing an operation without additional parameters (yeah, they could be used for a check, but it's not obligatory).